src/graph/util/ExpressionUtils.cpp

```diff
@@ -519,21 +556,41 @@ Expression *ExpressionUtils::rewriteRelExprHelper(const Expression *expr,
     // case Expression::Kind::kMultiply:
     // case Expression::Kind::kDivision:
     default:
-      LOG(FATAL) << "Unsupported expression kind: " << static_cast<uint8_t>(kind);
+      DLOG(ERROR) << "Unsupported expression kind: " << static_cast<uint8_t>(kind);
       break;
   }
 
   return rewriteRelExprHelper(root, relRightOperandExpr);
 }
 
 StatusOr<Expression *> ExpressionUtils::filterTransform(const Expression *filter) {
-  auto rewrittenExpr = const_cast<Expression *>(filter);
+  // If the filter contains more than one different LabelAttribute expr, this filter cannot be
+  // pushed down
+  auto propExprs = ExpressionUtils::collectAll(filter, {Expression::Kind::kLabelTagProperty});
+  // Deduplicate the list
+  auto dedupPropExprsList =
+      std::unordered_set<const Expression *>(propExprs.begin(), propExprs.end());
```

Comment:
should  de-duplicated the value not the pointer

Reply:
Nice catch